src/base/packable_object.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * packable_object.hpp
+ */
+
+#ifndef _PACKABLE_OBJECT_HPP_
+#define _PACKABLE_OBJECT_HPP_
+
+#ident "$Id$"
+
+#include "packer.hpp"
+#include <map>
+
+namespace cubpacking
+{
+
+  class packable_object
+  {
+    public:
+      virtual int pack (packer *serializator) = 0;
+      virtual int unpack (packer *serializator) = 0;
+
+      virtual bool is_equal (const packable_object *other) = 0;
```

Comment:
can't we add a '==' operator override instead?

Reply:
Having mixed C and C++ style, I think is best to avoid overloading operators, it may create confusion.
I also think the code is more readable with a named method.